src/main/java/gift/member/service/MemberService.java

```diff
@@ -19,19 +24,33 @@ public MemberService(MemberRepository memberRepository, JwtUtil jwtUtil) {
         this.jwtUtil = jwtUtil;
     }
 
-    public TokenDTO register(Member member) {
+    public Token register(Member member) {
+        MemberEntity existingMemberEntity = memberRepository.findByEmail(member.getEmail());
+        if (existingMemberEntity != null) {
+            throw new AlreadyExistsException("Alreay Exists Member");
+        }
         member.setPassword(PasswordUtil.encodePassword(member.getPassword()));
-        memberRepository.save(member);
-        return new TokenDTO(jwtUtil.generateToken(member));
+        memberRepository.save(dtoToEntity(member));
+
+        MemberEntity memberEntity = memberRepository.findByEmail(member.getEmail());
+        return new Token(jwtUtil.generateToken(entityToDto(memberEntity)));
     }
 
-    public TokenDTO login(Member member) {
-        Member existingMember = memberRepository.findByEmail(member.getEmail());
-        if (existingMember != null &&
-            PasswordUtil.decodePassword(existingMember.getPassword()).equals(member.getPassword())) {
-            return new TokenDTO(jwtUtil.generateToken(member));
+    public Token login(Member member) {
+        MemberEntity existingMemberEntity = memberRepository.findByEmail(member.getEmail());
+        if (existingMemberEntity != null &&
+            PasswordUtil.decodePassword(existingMemberEntity.getPassword()).equals(member.getPassword())) {
+            return new Token(jwtUtil.generateToken(member));
         }
```

Comment:
passwordUtil에 패스워드를 비교해주는 메서드를 만드는건 어떨까요?
```suggestion
        if (PasswordUtil.matches(member.getPassword(), existingMemberEntity.getPassword())) {
            return new Token(jwtUtil.generateToken(member));
        }
```